### DISCUSSION TITLE
How can I configure SSO in Abp.io to support Azure AD across multiple tenants that are not under the same Azure portal?

### DISCUSSION THREAD

**Discussion by vishnukumarspeehive** [abpframework/abp#19743] [General]
I am using the Abp.io 5.3 framework Angular non-layer application. Using the below code I've successfully implemented Azure AD Single Sign-On (SSO) for my application using the Abp.io 5.3 framework with Angular. 
![Question1](https://github.com/abpframework/abp/assets/167053378/d5701d59-12e8-4b59-8b32-fc519948546e)
However, I now need to extend SSO to include a new domain that isn't managed under my Azure portal. I do have the necessary credentials - client ID, secret key, and tenant ID - for this new tenant's Azure App.
Since these two tenants aren't managed under the same Azure portal, I'm unsure how to proceed with SSO. Is there a method to dynamically load OpenID Connect settings based on the domain name, allowing me to integrate SSO for this new domain alongside the existing implementation?

**Comment by maliming** [comment]
hi

The dynamic external login feature is available on abp commercial. It allows you to set different `ClientId,` `ClientSecret,` etc. for different tenants.

https://docs.abp.io/en/commercial/latest/modules/account#configure-the-provider
